leakcanary-android-core/src/main/java/leakcanary/internal/HeapAnalyzerService.kt

```diff
@@ -52,15 +53,20 @@ internal class HeapAnalyzerService : ForegroundService(
     // Since we're running in the main process we should be careful not to impact it.
     Process.setThreadPriority(Process.THREAD_PRIORITY_BACKGROUND)
     val heapDumpFile = intent.getSerializableExtra(HEAPDUMP_FILE_EXTRA) as File
+    val heapDumpDurationMillis = intent.getLongExtra(HEAPDUMP_DURATION_MILLIS, -1)
 
     val config = LeakCanary.config
     val heapAnalysis = if (heapDumpFile.exists()) {
       analyzeHeap(heapDumpFile, config)
     } else {
       missingFileFailure(heapDumpFile)
     }
+    val fullHeapAnalysis = when (heapAnalysis) {
+      is HeapAnalysisSuccess -> heapAnalysis.copy(dumpDurationMillis = heapDumpDurationMillis)
+      is HeapAnalysisFailure -> heapAnalysis.copy(dumpDurationMillis = heapDumpDurationMillis)
+    }
```

Comment:
👍

Reply:
To be honest, I wish smart cast would be smarter, but the sealed abstract class does not have `copy()` generated, and I have to downcast a `HeapAnalysis` to use the copy. This explains why the duplication body in different switch cases.